PyInstaller/hooks/hook-gevent.py

```diff
@@ -13,4 +13,7 @@
 
 excludedimports = ["gevent.testing", "gevent.tests"]
 
-datas, binaries, hiddenimports = collect_all('gevent')
+datas, binaries, hiddenimports = collect_all(
+    'gevent',
+    filter_submodules=lambda name: (
+        "gevent.testing" or "gevent.tests" not in name))
```

Comment:
This doesn't do what you think it does.
```python
>> "a" or "b" in ["x", "y", "z"]
"a"
>> bool("a" or "b" in ["x", "y", "z"])
True
```
So your `lambda name: ("gevent.testing" or "gevent.tests" not in name))` will always return `"gevent.testing"` which, after `bool` is applied, becomes `True`. And those testing modules will be included anyway. Try:
```suggestion
        "gevent.testing" not in name or "gevent.tests" not in name))
```

Reply:
Yeah my mistake - thanks @bwoodsend